e2e/terraform/provision-infra/userdata/windows-2022.ps1

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+<powershell>
+
+Set-StrictMode -Version latest
+$ErrorActionPreference = "Stop"
+
+$RunningAsAdmin = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")
+if (!$RunningAsAdmin) {
+  Write-Error "Must be executed in Administrator level shell."
+  exit 1
+}
+
+# -------------------------------------------
+# Disks and Directories
+
+# Bring ebs volume online with read-write access
+Get-Disk | Where-Object IsOffline -Eq $True | Set-Disk -IsOffline $False
+Get-Disk | Where-Object isReadOnly -Eq $True | Set-Disk -IsReadOnly $False
+
+New-Item -ItemType Directory -Force -Path C:\opt -ErrorAction Stop
+New-Item -ItemType Directory -Force -Path C:\opt\nomad
+New-Item -ItemType Directory -Force -Path C:\etc\nomad.d
+New-Item -ItemType Directory -Force -Path C:\tmp
+New-Item -ItemType Directory -Force -Path C:\opt\consul
+New-Item -ItemType Directory -Force -Path C:\etc\consul.d
+
+# -------------------------------------------
+# Install Consul Agent
+
+Set-Location C:\opt
+
+Try {
+    $releases = "https://releases.hashicorp.com"
+    $version = "1.21.1+ent"
```

Comment:
this might be neat to have as a var at the top of the script, if that's easy? (I never retain powershell learnings for very long)

I suppose there aren't really any other vars to call out as overall script "config", so meh.

Reply:
Yeah I was trying to keep the scope more local because I can't figure out whether `Try` actually introduces a new scope or not in the powershell docs.